### PR TITLE
Allow to deploy a Prometheus Operator ServiceMonitor

### PR DESCRIPTION
**Pull request by splattner**
# Pull Request

## Description of the change

This allows to deploy a Prometheus Operator ServiceMonitor for the metrics scraping.

## Benefits

`prometheus.io/scrape` and `prometheus.io/port` are not really used anymore and instead of manually creating a ServiceMonitor this chart should allow to create the ServiceMonitor

## Possible drawbacks

none that I am aware off

## Applicable issues

none

## Checklist 

- [x] DCO has been [signed off on the commit](https://docs.github.com/en/github/authenticating-to-github/signing-commits).
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] (optional) Variables are documented in the README.md

### CODE REVIEW

**Review by mrueg — APPROVED** [review]

File: charts/nextcloud/values.yaml [hunk 1]
```diff
@@ -459,6 +459,36 @@ metrics:
...
+    namespace: ""
+
+    ## @param metrics.serviceMonitor.jobLabel The name of the label on the target service to use as the job name in prometheus.
+    ##
+    jobLabel: ""
+
+    ## @param metrics.serviceMonitor.interval Interval at which metrics should be scraped
+    ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint
+    ##
+    interval: 30s
```
Comment (mrueg) [inline charts/nextcloud/values.yaml:480]: @splattner  can you fix the linting issue here?

Comment (splattner) [reply charts/nextcloud/values.yaml:480]: sure, I'll have a look again and will fix soon (was on holiday until today)

Comment (splattner) [reply charts/nextcloud/values.yaml]: I removed the space :+1:

**Review by mrueg — APPROVED** [review]

**Comment by splattner** [comment]
I bumped the chart version in my PR to pass the lint test again.

File: charts/nextcloud/templates/metrics-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "nextcloud.fullname" . }}
+  {{- if .Values.metrics.serviceMonitor.namespace }}
+  namespace: {{ .Values..metrics.serviceMonitor.namespace | quote }}
```
Comment (mrueg) [inline charts/nextcloud/templates/metrics-servicemonitor.yaml:7]:
```suggestion
  namespace: {{ .Values.metrics.serviceMonitor.namespace | quote }}
```

File: charts/nextcloud/templates/metrics-servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    app.kubernetes.io/name: {{ include "nextcloud.name" . }}
+    helm.sh/chart: {{ include "nextcloud.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/component: metrics
+    {{- if .Values.metrics.serviceMonitor.labels -}}
+    {{ toYaml .Values.metrics.serviceMonitor.labels | nindent 4 }}
+    {{- end -}}
+spec:
+  jobLabel: {{ .Values.server.metrics.serviceMonitor.jobLabel | quote }}
```
Comment (mrueg) [inline charts/nextcloud/templates/metrics-servicemonitor.yaml:21]:
```suggestion
  jobLabel: {{ .Values.metrics.serviceMonitor.jobLabel | quote }}
```

**Comment by alesc** [comment]
with only `metrics.serviceMonitor.enabled: true` configuration changed from default values.  

Helm upgrade : 
`Error: UPGRADE FAILED: YAML parse error on nextcloud/templates/metrics-servicemonitor.yaml: error converting YAML to JSON: yaml: line 11: mapping values are not allowed in this context` on helm upgrade.

Is there additional mandatory configration ?

**Comment by splattner** [comment]
let me have a look and fix this

**Comment by splattner** [comment]
this https://github.com/nextcloud/helm/pull/245 should fix the error

**Pull request merged**